js/TxUtils.js

```diff
@@ -74,13 +74,29 @@ function isValidOpenAliasAddress(address) {
     return true
 }
 
-function makeQRCodeString(addr, amount) {
+function makeQRCodeString(addr, amount, txDescription, recipientName) {
     var XMR_URI_SCHEME = "monero:"
     var XMR_AMOUNT = "tx_amount"
+    var XMR_RECIPIENT_NAME = "recipient_name"
+    var XMR_TX_DESCRIPTION = "tx_description"
     var qrCodeString =""
     qrCodeString += (XMR_URI_SCHEME + addr)
     if (amount !== undefined && amount !== ""){
       qrCodeString += ("?" + XMR_AMOUNT + "=" + amount)
     }
+    if (txDescription !== undefined && txDescription !== ""){
+        if (amount == ""){
+            qrCodeString += ("?" + XMR_TX_DESCRIPTION + "=" + txDescription)
```

Comment:
The "spec" says we have to replace spaces with `%20`: https://github.com/monero-project/monero/wiki/URI-Formatting

Reply:
You can use `encodeURI(txDescription)` and `encodeURI(recipientName)` here.